extension.html

```diff
@@ -0,0 +1,148 @@
+      </p>
+    </section>
+    <section>
+      <h2>Extensions to the <code>PointerEvent</code> interface</h2>
+      <p>
+        The following section describes extensions to the existing <a href="https://w3c.github.io/pointerevents/index.html#pointerevent-interface"><code>PointerEvent</code> interface</a>.
+      </p>
+      <div>
+          <pre class="idl">
+partial dictionary PointerEventInit {
+    sequence&lt;PointerEvent> coalescedEvents = [];
+};
+
+partial interface PointerEvent {
+    sequence&lt;PointerEvent> getCoalescedEvents();
+};
+          </pre>
+          <dl dfn-for="PointerEvent" link-for="PointerEvent">
+              <dt><dfn>getCoalescedEvents</dfn></dt>
+              <dd>
+                  <p>Returns a sequence of all <code>PointerEvents</code> that were coalesced into the received <code>pointermove</code> event. The following attributes of the coalesced events will always have the same value as the received event: <code>pointerId</code>, <code>pointerType</code>, <code>isPrimary</code>, and particularly <code>target</code>. But other attributes might be different. The events in the sequence will have increasing <a href="https://dom.spec.whatwg.org/#dom-event-timestamp"><code>timeStamps</code></a> ([[!WHATWG-DOM]]). So the first event will have the smallest <code>timeStamp</code>. The received event's attributes will be initalized in a way that is best representative of all the coalesced events. For example It's <code>timeStamp</code> will be equal to the last event's <code>timeStamp</code> in the sequence and it's <a href="https://www.w3.org/TR/pointerlock/#widl-MouseEvent-movementX">movementX</a> and <a href="https://www.w3.org/TR/pointerlock/#widl-MouseEvent-movementY">movementY</a> ([[!POINTERLOCK]]) will be the sum of that of all the coalesced events. None of the events in the sequence will have coalesced events and <code>getCoalescedEvents</code> returns an empty sequence for them. This API always returns at least one coalesced event for <code>pointermove</code> events and an empty list for other types of <code>PointerEvents</code>.</p>
+              </dd>
+          </dl>
+      </div>
+    </section>
+
+    <section>
+      <h2>Timing of coalesced event firing</h2>
+      <p>Coalescing events only happens for <code>pointermove</code> events. All the coalesced events should be fired before the next frame or before any <code>pointerup</code>, <code>pointerdown</code>, <code>pointercancel</code> events. Note that user agents don't need to hit-test all the coalesced events as all of them will have the same target as the dispatched event.</p>
+      <p>When the user agent dispatches any <code>PointerEvent</code> it MUST dispatch an event for every <code>pointerId</code> that has any coalesced events so far at that point with the correct relative ordering.</p>
```

Comment:
Here "has any coalesced events" may seem to include only pointermoves. Also "an event for every pointerId" is not a perfect description because your example below (which is great btw) has two events with id=1. What about this: "...it MUST dispatch all events held back so far for coalescing, regardless of their pointerIds. The ordering of all these dispatched events ...